test/interface/sparsediff_tests.jl

```diff
@@ -34,22 +47,54 @@ jac_sp = sparse(generate_sparsity_pattern(10))
 colors = repeat(1:3,10)[1:10]
 u0=[1.,2.,3,4,5,5,4,3,2,1]
 tspan=(0.,10.)
-odefun_sp= ODEFunction(f,colorvec=colors,jac_prototype=jac_sp)
-prob_sp = ODEProblem(odefun_sp,u0,tspan)
-prob_std = ODEProblem(f,u0,tspan)
-
-sol_sp=solve(prob_sp,Rodas5(autodiff=false),abstol=1e-10,reltol=1e-10)
-@test sol_sp.retcode==:Success#test sparse finitediff
-sol=solve(prob_std,Rodas5(autodiff=false),abstol=1e-10,reltol=1e-10)
-@test sol_sp.u[end]≈sol.u[end] atol=1e-10
-@test length(sol_sp.t)==length(sol.t)
-
-sol_sp=solve(prob_sp,Rodas5(autodiff=false))
-sol=solve(prob_std,Rodas5(autodiff=false))
-@test sol_sp.u[end]≈sol.u[end]
-@test length(sol_sp.t)==length(sol.t)
-
-sol_sp=solve(prob_sp,Rodas5())
-sol=solve(prob_std,Rodas5())
-@test sol_sp.u[end]≈sol.u[end]
-@test length(sol_sp.t)==length(sol.t)
\ No newline at end of file
+
+for f in [f_oop, f_ip]
+  odefun_std = ODEFunction(f)
+  prob_std = ODEProblem(odefun_std,u0,tspan)
+
+  for ad in [true, false]
+    for Solver in [Rodas5, Trapezoid, KenCarp4]
+      for tol in [nothing, 1e-10]
+        # @show f,ad,Solver,tol
+        sol_std=solve(prob_std,Solver(autodiff=ad),reltol=tol,abstol=tol)
+        @test sol_std.retcode==:Success
+        for (i,prob) in enumerate(map(f->ODEProblem(f,u0,tspan),
+                        [ODEFunction(f,colorvec=colors,jac_prototype=jac_sp),
+                         ODEFunction(f,jac_prototype=jac_sp),
+                         ODEFunction(f,colorvec=colors)
+                         ]))
+          isbroken = i==3 && (
+            (f, ad, tol) == (f_oop, true, nothing) ||
+            (f, ad, Solver, tol) == (f_oop, false, Trapezoid, nothing) ||
+            (f, Solver, tol) == (f_ip, Trapezoid, nothing) ||
+            (f, Solver) == (f_oop, Rodas5) ||
+            (f, Solver) == (f_ip, Rodas5) ||
+            (f, Solver) == (f_oop, KenCarp4) ||
+            (f, Solver) == (f_ip, KenCarp4)
+          )
```

Comment:
These failing tests need investigation.

Reply:
I wouldn't expect colors without sparsity to work.

Reply:
IMO, the color stuff should be independent of the matrix type of J.

Reply:
These are still "broken".